plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java

```diff
@@ -378,10 +381,21 @@ protected static String formatImageName(@Nonnull String repository, @Nonnull Str
 
   @Nonnull
   protected DockerClient openDockerClient() throws MojoExecutionException {
+    ContainerRegistryAuthSupplier authSupplier = null;
+    try {
+      authSupplier = ContainerRegistryAuthSupplier.forApplicationDefaultCredentials()
+          .build();
+      getLog().info("Using Google application credentials");
+    } catch (IOException ex) {
```

Comment:
the two methods in the `try` here can only throw `IOException`:

https://github.com/spotify/docker-client/blob/3a9f04d4d81dd2cfacf8f5bd139f007964d4bac6/src/main/java/com/spotify/docker/client/gcr/ContainerRegistryAuthSupplier.java#L99

https://github.com/spotify/docker-client/blob/3a9f04d4d81dd2cfacf8f5bd139f007964d4bac6/src/main/java/com/spotify/docker/client/gcr/ContainerRegistryAuthSupplier.java#L160

it is the methods called at build-/pull-time that might throw `DockerException`